File: BloodFloater/src/BloodFloater.Web.UI/wwwroot/src/pages/profile/profile.service.ts
import { Injectable, Optional } from '@angular/core';
import { Profile } from '../../app/models';
import { IBaseService, BaseService } from '../../app/services/base.service';
import { Http, URLSearchParams } from '@angular/http';

export interface IProfileService extends IBaseService<Profile> { }

@Injectable()
export class ProfileService extends BaseService<Profile> implements IProfileService {  

    constructor(public http: Http){
        super(http, "profiles");
    }      
}

File: BloodFloater/src/BloodFloater.Services.Impl/ProfileService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using BloodFloater.DAL;
using BloodFloater.Models;

namespace BloodFloater.Services.Impl
{
    public class ProfileService : BaseService<Profile, ViewModels.Profile>, IProfileService
    {
        public ProfileService(IProfileRepository repository) : base(repository)
        {
        }
    }
}
